src/cli/api/api.ts

```diff
@@ -89,54 +92,105 @@ export async function fetchTrackingPlan(options: {
   email: string;
   APIVersion: string;
 }): Promise<RudderAPI.TrackingPlan> {
-  let url =
-    options.APIVersion === 'v1' ? `trackingplans/${options.id}` : `tracking-plans/${options.id}`;
-  url = options.version ? `${url}?version=${options.version}` : url;
-  const response = await apiGet<RudderAPI.GetTrackingPlanResponse>(
+  let response: RudderAPI.GetTrackingPlanResponse;
+
+  const { id, version, token, email } = options;
+
+  switch (options.APIVersion) {
+    case 'v1':
+      response = await _fetchTrackingPlanV1(id, version, token, email);
+      break;
+
+    case 'v2':
+      response = await _fetchTrackingPlanV2(id, version, token, email);
+      break;
+
+    default:
+      throw new Error(`Invalid API version: ${options.APIVersion}`);
+  }
+
+  return sanitizeTrackingPlan(response);
+}
+
+async function _fetchTrackingPlanV1(
+  id: string,
+  version: number | undefined,
+  token: string,
+  email: string,
+): Promise<RudderAPI.GetTrackingPlanResponse> {
+  const url = version ? `trackingplans/${id}?version=${version}` : `trackingplans/${id}`;
+
+  const response = await apiGet<RudderAPI.GetTrackingPlanResponse>(url, token, email);
+  response.create_time = new Date(response.create_time);
+  response.update_time = new Date(response.update_time);
+
+  return response;
+}
+
+async function _fetchTrackingPlanV2(
+  id: string,
+  version: number | undefined,
+  token: string,
+  email: string,
+): Promise<RudderAPI.GetTrackingPlanResponse> {
+  const url = version ? `tracking-plans/${id}?version=${version}` : `tracking-plans/${id}`;
+
+  const response = await apiGet<RudderAPI.GetTrackingPlanResponse>(url, token, email);
+  response.createdAt = new Date(response.createdAt);
+  response.updatedAt = new Date(response.updatedAt);
+
+  const rules = await _fetchTrackingPlanV2RulesByPage(id, token, email);
+  response.rules = {
+    events: rules,
+  };
+
+  return response;
+}
+
+async function _fetchTrackingPlanV2RulesByPage(
+  id: string,
+  token: string,
+  email: string,
+  page = 1,
+): Promise<RudderAPI.RuleMetadata[]> {
+  const url = `tracking-plans/${id}/events/${page}`;
+
+  const trackingplanEvents = await apiGet<RudderAPI.GetTrackingPlanEventsResponse>(
```

Comment:
If events are empty in the subsequent pages then we need to exit the loop for pagination